docs: update picking sources and unify with patching rules (#14377)
conan-io/conan-center-index: docs/developing_recipes_locally.md

To test with the same environment, the [build images](supported_platforms_and_configurations.md#build-images) are available.
Instructions for using these images can be found in [Testing more environments](#testing-more-environments) section.

> In ConanCenterIndex, the most common failure point is upstream build script that are tailored to their specific use cases.

Reviewer: ```suggestion
In ConanCenterIndex, the most common failure point is upstream build script that is tailored to its specific use cases.
```
